### PR TITLE
feat(dialog): better handle of dialog interactions

### LINKED ISSUE #141: small add to library bug
This isn't a big deal but whenever you click add to library and click cancel, it adds the game to the library even though no executable is assigned to it. Not sure if this is intended or not.

### LINKED ISSUE #240: [BUG] Botão de adicionar à biblioteca
[BUG] Ao clicar no botão de adicionar à biblioteca, a janela para seleção de pasta, abre normalmente, porém, caso a mesma seja fechada (sem que uma pasta seja selecionada), os botões mudam 
De:
image
Para:
image
Sem que nenhum jogo tenha sido baixado ou selecionado.

Passo a passo:
1. Página do jogo, acessado a partir do botão de sugestões.
image
2. Pasta do SO aberta para seleção. Cliquei em "cancelar":
image
3. Botões alterados:
image

É isso!

### LINKED ISSUE #277: [BUG] Can't select .exes to add to liabry on Steam Deck
**Describe the bug**
The selector for the game on Steam Deck is not recognising .exe types of files, so I can't add them into the launcher

**To Reproduce**

1. Add any game to your libary (not installed from Hydra)
2. Click on it and click on play
3. Try to go to game's folder and apply the .exe file to it
4. Can't apply

**Expected behavior**
Reaching the folder and being able to select .exe file

**Screenshots**
image
image

**Additional context**

- OS: Steam Deck
- Hydra Version: 1.1.0

### PR DESCRIPTION
**Pull request by scrlkx**
I have made a few improvements on using and handling dialogs.

1. Fixed the issue that make calls not wait for the selection dialog (#240, #141 and #232).
2. Improved the executable files filter to allow more types of files. At this point, we can just select `.exe` files on Windows and no types of files on Linux/MacOS (#277).
3. Added titles for all dialogs.

Fixes #240 
Fixes #141
Fixes #277

### CODE REVIEW

File: src/renderer/src/pages/game-details/hero/hero-panel-actions.tsx [hunk 1]
```diff
@@ -43,23 +43,22 @@ export function HeroPanelActions({
   const { t } = useTranslation("game_details");
 
   const selectGameExecutable = async () => {
-    return window.electron
-      .showOpenDialog({
-        properties: ["openFile"],
-        filters: [
-          {
-            name: "Game executable",
-            extensions: window.electron.platform === "win32" ? ["exe"] : [],
```
Comment (zamitto) [inline src/renderer/src/pages/game-details/hero/hero-panel-actions.tsx:52]: having an empty list on the filter actually makes the file selection show every possible file. this is not the cause of the bug

Comment (zamitto) [reply src/renderer/src/pages/game-details/hero/hero-panel-actions.tsx:52]: you can test it. Run the app with the "exe" filter, and then run it again with the empty list

Comment (scrlkx) [reply src/renderer/src/pages/game-details/hero/hero-panel-actions.tsx:52]: I did tested and it's not working. At least on Linux `[]` means no files at all. When I change to `['*']` it goes like a charm. There is a mention about it in the docs ([see](https://www.electronjs.org/docs/latest/api/dialog#methods)), maybe you want to checkout.

I even record a video, just in case.

https://github.com/hydralauncher/hydra/assets/22862212/5fadcc98-b35d-4d9a-830c-156e3bcb8e12

Comment (scrlkx) [reply src/renderer/src/pages/game-details/hero/hero-panel-actions.tsx:52]: Note that the video was record using the code from `main`.

Comment (scrlkx) [reply src/renderer/src/pages/game-details/hero/hero-panel-actions.tsx:52]: Btw, the video also shows that #240 and #141 got fixed.

Comment (zamitto) [reply src/renderer/src/pages/game-details/hero/hero-panel-actions.tsx:52]: Oh, I see. So I think a better modification would to set "*" for non-windows systems, and keep ".exe" for windows

Comment (scrlkx) [reply src/renderer/src/pages/game-details/hero/hero-panel-actions.tsx:52]: Have a "All Files (*)" can be usefull for cases like #302. What do you think?

Comment (zamitto) [reply src/renderer/src/pages/game-details/hero/hero-panel-actions.tsx:52]: I see your point, but for now I'd rather we keep only the .exe for Windows and think better how to solve #302 on another PR

Comment (scrlkx) [reply src/renderer/src/pages/game-details/hero/hero-panel-actions.tsx:52]: Ok. I changed it.

```typescript
    const { canceled, filePaths } = await window.electron.showOpenDialog({
      title: t("game_executable_selection.title"),
      properties: ["openFile"],
      filters: [
        {
          name: window.electron.platform === "win32" ? t("game_executable_selection.executable_files") : t("game_executable_selection.all_files"),
          extensions: window.electron.platform === "win32" ? ["exe"] : ["*"],
        },
      ],
    });
```

**Comment by scrlkx** [comment]
@zamitto and @hydralauncher should I update this or are we going do discard it?

File: src/main/events/misc/show-open-dialog.ts [hunk 2]
```diff
@@ -7,7 +7,7 @@ const showOpenDialog = async (
   options: Electron.OpenDialogOptions
 ) => {
   if (WindowManager.mainWindow) {
-    return dialog.showOpenDialog(WindowManager.mainWindow, options);
+    return await dialog.showOpenDialog(WindowManager.mainWindow, options);
```
Comment (henriqueleite42) [inline src/main/events/misc/show-open-dialog.ts:10]: Unnecessary change, already returning a promise

**Comment by henriqueleite42** [comment]
@zamitto This was a nice PR, please give it some attention :pray:
